### ISSUE TITLE
Support empty string as roslaunch argument (programmatic roslaunch API)

### ISSUE THREAD

**Issue by luisrayas3** [ros/ros_comm#2104]
Note: I have seen #1447 but this deals with the command-line API. I have a similar issue when using the python API and nothing seems to work. Here is a minimal example of the problem:

Consider:

```xml
<?xml version="1.0"?>
<!-- File 'test.launch' in package 'baz' -->
<launch>
  <arg name="foo" />
  <param name="FOO" value="$(arg foo) is NOT empty!" if="$(eval foo != '')" />
  <param name="FOO" value="$(arg foo) is empty" if="$(eval foo == '')" />
</launch>
```

Now a launch python script:

```python
import roslaunch

file = roslaunch.rlutil.resolve_launch_arguments(['baz', 'test.launch'])[0]

# None of these work:
args = ["foo:="]
#args = ["foo:=''"]
#args = ['foo:=""']

run_id = roslaunch.rlutil.get_or_generate_uuid(None, False)
parent = roslaunch.parent.ROSLaunchParent(run_id, [(file, args)])
parent.start()
```

If I use the uncommented version of `args`, I get the following:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/ros/melodic/lib/python2.7/dist-packages/roslaunch/parent.py", line 289, in start
    self._start_infrastructure()
  File "/opt/ros/melodic/lib/python2.7/dist-packages/roslaunch/parent.py", line 238, in _start_infrastructure
    self._load_config()
  File "/opt/ros/melodic/lib/python2.7/dist-packages/roslaunch/parent.py", line 144, in _load_config
    roslaunch_strs=self.roslaunch_strs, verbose=self.verbose)
  File "/opt/ros/melodic/lib/python2.7/dist-packages/roslaunch/config.py", line 462, in load_config_default
    raise RLException(e)
roslaunch.core.RLException: [/.../baz/launch/test.launch] requires the 'foo' arg to be set
```

whereas if I use either of the commented out lines instead, it does succeed in launching, but I get:

```
PARAMETERS
 * /FOO: "" is NOT empty!
```

In other words, it seems it is impossible to send an empty string explicitly as the value of an argument. Either it treats it as not sent or as the string representation of an empty string. This is OK in cases where there is a `default=""` for the `arg` but is a problem if one wants to avoid a default (e.g. to fail if it was not explicitly provided) or, more seriously, if the default argument is not the empty string but passing in an empty string is desired.

**Comment by sloretz** [comment]
ROS 1 is end-of-life (EOL) as of today, May 31st 2025. I am archiving this repository because:

* it only supports ROS 1
* it isn't needed anymore in ROS 2

[If you still rely on ROS 1, read this page to learn about your options.](https://bit.ly/NoeticEOL)
